sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -76,6 +76,24 @@ abstract class DataType {
     DataType.equalsIgnoreNullability(this, other)
 
   /**
+   * Check if `this` is a primitive data type.
+   */
+  def isPrimitive: Boolean = {
```

Comment:
Can't we directly check whether the data type is ArrayType using datatype.isInstanceOf [ArrayType] or wouldn't it be better to directly check if the data cell is of type Sequence using .isInstanceOf [ Seq[ _ ] ] ?